src/cli.rs

```diff
@@ -289,179 +485,15 @@ impl Config {
-
-        let ascii_language = matches
-            .value_of("ascii-language")
-            .map(|ascii_language| Language::from_str(&ascii_language.to_lowercase()).unwrap());
-
-        let ascii_colors = if let Some(values) = matches.values_of("ascii-colors") {
-            values.map(String::from).collect()
-        } else {
-            Vec::new()
-        };
-
-        let text_colors = if let Some(values) = matches.values_of("text-colors") {
-            values.map(String::from).collect()
-        } else {
-            Vec::new()
-        };
-
-        let number_of_authors: usize = matches.value_of("authors-number").unwrap().parse()?;
-
-        let ignored_directories =
-            if let Some(user_ignored_directories) = matches.values_of("exclude") {
-                user_ignored_directories.map(String::from).collect()
-            } else {
-                Vec::new()
-            };
-
-        let bot_regex_pattern = matches.is_present("no-bots").then(|| {
-            matches
-                .value_of("no-bots")
-                .map_or(Regex::from_str(r"\[bot\]").unwrap(), |s| {
-                    Regex::from_str(s).unwrap()
-                })
-        });
-
-        let language_types: Vec<LanguageType> = matches
-            .values_of("type")
-            .unwrap()
-            .map(|t| LanguageType::from_str(t).unwrap())
-            .collect();
-
-        Ok(Config {
-            repo_path,
-            ascii_input,
-            ascii_language,
-            ascii_colors,
-            disabled_fields,
-            no_bold,
-            image,
-            image_backend,
-            image_color_resolution,
-            no_merges,
-            no_color_palette,
-            number_of_authors,
-            ignored_directories,
-            bot_regex_pattern,
-            print_languages,
-            print_package_managers,
-            output,
-            true_color,
-            art_off,
-            text_colors,
-            iso_time,
-            show_email,
-            include_hidden,
-            language_types,
-        })
-    }
-}
-
-pub fn print_supported_languages() -> Result<()> {
-    for l in Language::iter() {
-        println!("{}", l);
-    }
-
-    Ok(())
-}
-
-pub fn print_supported_package_managers() -> Result<()> {
-    for p in PackageManager::iter() {
-        println!("{}", p);
-    }
-
-    Ok(())
-}
-
-pub fn is_truecolor_terminal() -> bool {
-    env::var("COLORTERM")
-        .map(|colorterm| colorterm == "truecolor" || colorterm == "24bit")
-        .unwrap_or(false)
-}
-
-pub fn get_git_version() -> String {
-    let version = Command::new("git").arg("--version").output();
-
-    match version {
-        Ok(v) => String::from_utf8_lossy(&v.stdout).replace('\n', ""),
-        Err(_) => String::new(),
-    }
+        .arg(
+            Arg::new("completion")
+                .long("completion")
+                .possible_values(possible_shells)
```

Comment:
```suggestion
                .possible_values(Shell::possible_values())
```

More readable IMO

Reply:
Yeah, this was from when I was experimenting with the lifetimes of the Args.